### DISCUSSION TITLE
Restart Rocket in Docker container

### DISCUSSION THREAD

**Discussion by Giga-Pudding** [dani-garcia/vaultwarden#4065] [General]
Hey folks,

I want to restart Rocket inside the Vaultwarden Docker Container for automated certificate renewal.
I thought of using `docker exec vaultwarden ...` but i don't know how exactly. Can anybody give me a hint?

I don't want to restart the whole container, since other components like the database do not depend on the SSL certificate.

BR,
Giga-Pudding

**Comment by RealOrangeOne** [comment]
If you're using a database like PostgreSQL (ie in its own container), then restarting just the vaultwarden container won't restart the database.  If you're using SQLite, then that's part of the same vaultwarden binary, so there is both nothing to restart, _and_ it by definition needs to be restarted with vaultwarden.

Either way, restarting the vaultwarden container makes the most sense, and is really your only option, since the container only contains the vaultwarden binary anyway. If having the small amount (a couple seconds) of downtime for a container restart is really that huge of an issue, then extracting TLS termination to another reverse proxy which supports reloading certificates (eg `nginx`) would be best.

**Comment by Giga-Pudding** [comment]
What happens, if a container restart command is issued exactly in the moment, while data is written to the database?
Is docker waiting somehow, or is the process terminated?

**Reply by BlackDex** [reply]
If you do a nice stop instead of a kill, then Vaultwarden will shutdown nicely

**Reply by Giga-Pudding** [reply]
how is a nice stop performed?

**Reply by BlackDex** [reply]
`docker stop vaultwarden`
